### PR TITLE
Implement PSR-18

### PR DESCRIPTION
**Pull request by Nyholm**
I drop old PHP versions. Update PHPUnit and Implement PSR-18

This will replace #2122

### CODE REVIEW

**Comment by gmponos** [comment]
Is this PR only for PSR-18.. ? For instance there is one more deprecation on one exception... We will do the separately after 6.5 tag?

File: .travis.yml [hunk 1]
```diff
@@ -2,35 +2,31 @@ language: php
...
+env:
+  COMPOSER_FLAGS="--no-interaction --prefer-source --dev"
+
 php:
-  - 5.6
-  - 7.0
-  - 7.1
   - 7.2
   - 7.3
   - 7.4snapshot
```
Comment (gmponos) [inline .travis.yml]: I think snapshot is not needed anymore? Am I right?

Comment (Nyholm) [reply .travis.yml]: It was needed 4 days ago

**Comment by Nyholm** [comment]
I will release 6.5 and then start working on 7.0. So this will be the first PR I merge after 6.5 is released.

File: src/Client.php [hunk 2]
```diff
@@ -130,6 +131,18 @@ public function send(RequestInterface $request, array $options = [])
         return $this->sendAsync($request, $options)->wait();
     }
 
+    /**
+     * The HttpClient PSR (PSR-18) specify this method.
+     */
+    public function sendRequest(RequestInterface $request): ResponseInterface
```
Comment (gmponos) [inline src/Client.php]: Thinking out loud here.. without having the answer on my head.. 

Should this be final?

File: .travis.yml (hunk 1 above)
Comment (Nyholm) [reply .travis.yml]: Nice!

**Comment by Nyholm** [comment]
This is just the first PR. 
We should add more integration tests and add PHP7.2 code. But one step at the time

File: README.md [hunk 3]
```diff
@@ -79,12 +79,15 @@ composer update
...
 [guzzle-3-repo]: https://github.com/guzzle/guzzle3
 [guzzle-4-repo]: https://github.com/guzzle/guzzle/tree/4.x
 [guzzle-5-repo]: https://github.com/guzzle/guzzle/tree/5.3
 [guzzle-6-repo]: https://github.com/guzzle/guzzle
+[guzzle-7-repo]: https://github.com/guzzle/guzzle
 [guzzle-3-docs]: http://guzzle3.readthedocs.org
-[guzzle-5-docs]: http://guzzle.readthedocs.org/en/5.3/
-[guzzle-6-docs]: http://guzzle.readthedocs.org/en/latest/
+[guzzle-5-docs]: http://docs.guzzlephp.org//en/5.3/
+[guzzle-6-docs]: http://docs.guzzlephp.org//en/6.5/
```
Comment (sagikazarmark) [inline README.md]:
```suggestion
[guzzle-6-docs]: http://docs.guzzlephp.org//en/6.x/
```

If the branch is  6.x.

File: src/functions.php [hunk 4]
```diff
@@ -133,7 +133,7 @@ function default_user_agent()
     static $defaultAgent = '';
 
     if (!$defaultAgent) {
-        $defaultAgent = 'GuzzleHttp/' . Client::VERSION;
+        $defaultAgent = 'GuzzleHttp/Guzzle';
```
Comment (sagikazarmark) [inline src/functions.php]: Maybe add `7.x` here?

File: README.md (hunk 3 above)
Comment (Nyholm) [reply README.md]: I created a 6.5 branch already. 

My idea is that Guzzle 7 is a so easy upgrade so there is no reason to add more features to 6.x

File: src/functions.php (hunk 4 above)
Comment (Nyholm) [reply src/functions.php]: Then we still need to worry about the version. For what benefit?

Comment (GrahamCampbell) [reply src/functions.php]: You could always do the trick php-cs-fixer does to detect its version?

Comment (GrahamCampbell) [reply src/functions.php]: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.16.1/src/ToolInfo.php#L61-L72

**Comment by Nyholm** [comment]
Thank you for the reviews.

File: src/functions.php (hunk 4 above)
Comment (Nyholm) [reply src/functions.php]: That is pretty cool. But Im not sure we want the version at all. 

Anyhow. A followup PR may fix this.

**Pull request merged**